### PR TITLE
Theme switch

### PR DESCRIPTION
**Pull request by bibash28**
theme switch with testing

### CODE REVIEW

File: lib/app/app_widget.dart [hunk 1]
```diff
@@ -40,54 +68,38 @@ class _AppWidgetState extends State<AppWidget> {
...
-                    },
-                    initialRoute: '/splash',
-                    theme: _themeData,
-                    localizationsDelegates: [
-                      AppLocalizations.delegate,
-                      GlobalMaterialLocalizations.delegate,
-                      GlobalWidgetsLocalizations.delegate,
-                      GlobalCupertinoLocalizations.delegate,
-                    ],
-                    supportedLocales: const <Locale>[
```
Comment (hawkbee1) [inline lib/app/app_widget.dart:76]: We don't need to specify the supported locales anymore ?

Comment (bibash28) [reply lib/app/app_widget.dart:76]: main.dart
  supportedLocales: AppLocalizations.supportedLocales,
  
  app_localisation.dart
  static const List<Locale> supportedLocales = <Locale>[
    Locale('de'),
    Locale('en'),
    Locale('es'),
    Locale('fr'),
    Locale('it')
  ];
  
  This includes all the supported languages.

File: lib/app/pages/splash.dart [hunk 2]
```diff
@@ -11,11 +11,16 @@ import 'package:talao/app/shared/ui/ui.dart';
 import 'package:talao/app/shared/widget/base/page.dart';
 import 'package:talao/app/shared/widget/brand.dart';
 import 'package:talao/deep_link/deep_link.dart';
+import 'package:talao/theme/theme.dart';
```
Comment (hawkbee1) [inline lib/app/pages/splash.dart:13]: [{
	"resource": "/home/rbarbe/Documents/flutter/spruce/spruceid/credible/lib/app/pages/splash.dart",
	"owner": "_generated_diagnostic_collection_name_#0",
	"code": {
		"value": "unused_import",
		"target": {
			"$mid": 1,
			"external": "https://dart.dev/diagnostics/unused_import",
			"path": "/diagnostics/unused_import",
			"scheme": "https",
			"authority": "dart.dev"
		}
	},
	"severity": 2,
	"message": "Unused import: 'package:talao/app/shared/ui/ui.dart'.\nTry removing the import directive.",
	"source": "dart",
	"startLineNumber": 10,
	"startColumn": 8,
	"endLineNumber": 10,
	"endColumn": 45
}]

**Review by hawkbee1 — CHANGES_REQUESTED** [review]
Some tests are failing: /home/rbarbe/Documents/flutter/spruce/spruceid/credible/test/app/theme/cubit/theme_cubit_test.dart
no big deal :)

File: test/app/pages/profile/pages/theme_test.dart [hunk 3]
```diff
@@ -0,0 +1,106 @@
+import 'package:bloc_test/bloc_test.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mocktail/mocktail.dart';
+import 'package:talao/app/pages/profile/pages/theme.dart';
+import 'package:talao/theme/cubit/theme_cubit.dart';
+
+import '../../../../helper/pump_app.dart';
```
Comment (hawkbee1) [inline test/app/pages/profile/pages/theme_test.dart:8]: please use full path here. That's just for consistency with the rest of the code

File: test/app/pages/profile/profile_test.dart [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mocktail/mocktail.dart';
+import 'package:talao/app/pages/profile/blocs/profile.dart';
+import 'package:talao/app/pages/profile/models/profile.dart';
+import 'package:talao/app/pages/profile/pages/theme.dart';
+import 'package:talao/app/pages/profile/profile.dart';
+import 'package:bloc_test/bloc_test.dart';
+import 'package:talao/theme/cubit/theme_cubit.dart';
+
+import '../../../helper/pump_app.dart';
```
Comment (hawkbee1) [inline test/app/pages/profile/profile_test.dart:12]: please use full path here. That's just for consistency with the rest of the code

File: lib/app/app_widget.dart [hunk 5]
```diff
@@ -12,25 +13,52 @@ import 'package:flutter_localizations/flutter_localizations.dart';
...
+            context.read<ScanBloc>(),
+            context.read<QueryByExampleCubit>(),
+          ),
+        ),
+        BlocProvider<ProfileBloc>(create: (context) => ProfileBloc()),
+        BlocProvider<DIDBloc>(create: (context) => DIDBloc()),
+      ],
+      child: MaterialAppDefinition(),
+    );
+    ;
```
Comment (hawkbee1) [inline lib/app/app_widget.dart]: this ; is dead code

**Review by bibash28 — COMMENTED** [review]
Yes, this can also be done. I am OK with this.

File: test/app/pages/profile/pages/theme_test.dart (hunk 3 above)
Comment (bibash28) [reply test/app/pages/profile/pages/theme_test.dart:8]: ok

**Comment by bibash28** [comment]
I will also look at the testing part of theme_cubit...

**Review by hawkbee1 — APPROVED** [review]

**Pull request merged**